tests/test_config_image.py

```diff
@@ -31,9 +33,612 @@
+            'random_seed' : 1234,
+            'draw_method' : 'phot'
+        },
+        'gal' : {
+            'type' : 'Gaussian',
+            'sigma' : 1.7,
+            'flux' : 100,
+        }
+    }
+
+    # First the simple config written above
+    ud = galsim.UniformDeviate(1234 + 1)
+    gal = galsim.Gaussian(sigma=1.7, flux=100)
+    im1a = gal.drawImage(scale=1, method='phot', rng=ud)
+    im1b = galsim.config.BuildImage(config)
+    np.testing.assert_array_equal(im1b.array, im1a.array)
+
+    # Use a non-default number of photons
+    del config['_copied_image_keys_to_stamp']
+    config['image']['n_photons'] = 300
+    ud.seed(1234 + 1)
+    im2a = gal.drawImage(scale=1, method='phot', n_photons=300, rng=ud)
+    im2b = galsim.config.BuildImage(config)
+    print('image = ',config['image'])
+    print('stamp = ',config['stamp'])
+    np.testing.assert_array_equal(im2b.array, im2a.array)
+
+    # Allow the flux to vary as a Poisson deviate even though n_photons is given
+    del config['_copied_image_keys_to_stamp']
+    config['image']['poisson_flux'] = True
+    ud.seed(1234 + 1)
+    im3a = gal.drawImage(scale=1, method='phot', n_photons=300, rng=ud, poisson_flux=True)
+    im3b = galsim.config.BuildImage(config)
+    np.testing.assert_array_equal(im3b.array, im3a.array)
+
+    # If max_extra_noise is given with n_photons, then ignore it.
+    del config['_copied_image_keys_to_stamp']
+    config['image']['max_extra_noise'] = 0.1
+    im3c = galsim.config.BuildImage(config)
+    np.testing.assert_array_equal(im3c.array, im3a.array)
+
+    # Although with a logger, it should give a warning.
+    with CaptureLog() as cl:
+        im3d = galsim.config.BuildImage(config, logger=cl.logger)
+    assert "ignoring 'max_extra_noise'" in cl.output
+
+    # Without n_photons, it should work.  But then, we also need a noise field
+    # So without the noise field, it will raise an exception.
+    del config['image']['n_photons']
+    del config['stamp']['n_photons']
+    try:
+        np.testing.assert_raises(AttributeError, galsim.config.BuildImage, config)
+    except ImportError:
+        pass
+
+    # Using this much extra noise with a sky noise variance of 50 cuts the number of photons
+    # approximately in half.
+    print('N,g without extra_noise: ',gal._calculate_nphotons(0, False, 0, None))
+    print('N,g with extra_noise: ',gal._calculate_nphotons(0, False, 5, None))
+    config['image']['noise'] = { 'type' : 'Gaussian', 'variance' : 50 }
+    ud.seed(1234 + 1)
+    im4a = gal.drawImage(scale=1, method='phot', max_extra_noise=5, rng=ud, poisson_flux=True)
+    im4a.addNoise(galsim.GaussianNoise(sigma=math.sqrt(50), rng=ud))
+    im4b = galsim.config.BuildImage(config)
+    np.testing.assert_array_equal(im4b.array, im4a.array)
+
+@timer
+def test_reject():
+    """Test various ways that objects can be rejected.
+    """
+    # Make a custom function for rejecting COSMOSCatalog objects that use Sersics with n > 2.
```

Comment:
I don't think it's _more_ helpful.  I just think it's another helpful thing.  They showcase different functionality (i.e., the one in cgc.yaml is based on properties of the drawn image whereas the one in this test lets you really dig into the COSMOSCatalog, and I don't find that trivial/obvious, personally).